scripts/generate-cluster-creds.sh

```diff
@@ -89,14 +89,25 @@ EOF
     rm kubeconfig.yaml
   fi
 
-  output=$(yq e -o json '.k8s_config' /tmp/temp_k8sconfig.yaml | jq -r -M -c .)
+  yaml_creds=$(yq '.k8s_config' /tmp/temp_k8sconfig.yaml)
+  json_creds=$(yq e -o json '.k8s_config' /tmp/temp_k8sconfig.yaml | jq -r -M -c .)
+
   # NOTE: Write to ci/ files as these files will be used in ci tests! Consider looping through all files
   # in ci dir
   for APP in merlin turing
   do
-    # Write to app chart and caraml chart ci-values.yaml
-    output="$output" yq ".environmentConfigs[0] *= load(\"/tmp/temp_k8sconfig.yaml\") | .imageBuilder.k8sConfig |= strenv(output)" -i "${SCRIPT_DIR}/../charts/${APP}/ci/ci-values.yaml"
-    output="$output" yq ".${APP}.environmentConfigs[0] *= load(\"/tmp/temp_k8sconfig.yaml\") | .${APP}.imageBuilder.k8sConfig |= strenv(output)" -i "${SCRIPT_DIR}/../charts/caraml/ci/ci-values.yaml"
+    # Write to Merlin/ Turing image builder app chart values
+    if [ $APP == "merlin" ]; then
+      yaml_creds="$yaml_creds" yq ".imageBuilder.k8sConfig |= env(yaml_creds)" -i "${SCRIPT_DIR}/../charts/${APP}/ci/ci-values.yaml"
+    else
+      json_creds="$json_creds" yq ".imageBuilder.k8sConfig |= strenv(json_creds)" -i "${SCRIPT_DIR}/../charts/${APP}/ci/ci-values.yaml"
+    fi
+    # Write to Merlin/ Turing env configs app chart values
+    yq ".environmentConfigs[0] *= load(\"/tmp/temp_k8sconfig.yaml\")" -i "${SCRIPT_DIR}/../charts/${APP}/ci/ci-values.yaml"
+
+    # TODO: TO REMOVE AFTER MERLN CHART HAS BEEN UPDATED
+    # Use JSON creds for Merlin in the overarching CaraML chart values as its chart is not merged yet
+    json_creds="$json_creds" yq ".${APP}.environmentConfigs[0] *= load(\"/tmp/temp_k8sconfig.yaml\") | .${APP}.imageBuilder.k8sConfig |= strenv(json_creds)" -i "${SCRIPT_DIR}/../charts/caraml/ci/ci-values.yaml"
```

Comment:
Since the overarching CaraML chart will automatically be tested due to its Merlin dependency, and because the Merlin chart that will be installed when testing the CaraML chart will be the one that's published, its app version will be an old app version that has not been published yet (i.e. does not contain the necessary refactoring to accept the yaml credentials). Thus we still need to parse the image builder cluster credentials to the Merlin configs in the overall Helm chart values file as a json string.